cmd/rook/rook/rook.go

```diff
@@ -181,3 +182,35 @@ func GetOperatorBaseImageCephVersion(context *clusterd.Context) (string, error)
 
 	return output, nil
 }
+
+// GetInternalOrExternalClient will get a Kubernetes client interface from the KUBECONFIG variable
+// if it is set, or from the operator pod environment otherwise.
+func GetInternalOrExternalClient() kubernetes.Interface {
+	kubeconfig := os.Getenv("KUBECONFIG")
+
+	var (
+		config *rest.Config
+		err    error
+	)
+
+	if kubeconfig != "" {
+		for _, kConf := range strings.Split(kubeconfig, ":") {
+			config, err = clientcmd.BuildConfigFromFlags("", kConf)
+			if err != nil {
+				break
+			}
+		}
+	} else {
+		config, err = rest.InClusterConfig()
```

Comment:
If KUBECONFIG is set, we don't want to fall back to this if the KUBECONFIG fails to load, right? That looks like the behavior, just confirming.

Reply:
Right. I think that behavior makes sense. KUBECONFIG shouldn't be set in the container, so I don't see a reason to fall back.